ls1intum/Artemis
# `General`: Improve navigation in course management

### Checklist
#### General

#### Client

### Motivation and Context


Before the changes, it was not so easy to navigate the course management.
Now, it is easy to go from one page to another without having to go back to the course detail page.
Closes #6187 

### Description

- Added new component: course-management-tab-bar
- Implemented this bar on all pages that the bar links to with one exception. Communication does not have the new course-management-tab-bar since it is not part of the course management but of course overview.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor

1. Log in to Artemis
2. Navigate to Course Management
3. Click on a course
4. Navigate to all pages the buttons link to. While doing so, verify that the buttons and links are loaded correctly and link to the correct page
5. Verify that other links still work as well

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-management-tab-bar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4747-38/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts.html) | 96.49% | ✅ |
| [course-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4747-38/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/detail/course-detail.component.ts.html) | 91.48% | ✅ |




### Screenshots


Before changes:
![7huwly](https://user-images.githubusercontent.com/47261058/231386201-fd7ef448-30af-4477-8e72-6537d5a04fba.gif)
After changes:
![7huwqc](https://user-images.githubusercontent.com/47261058/231386125-8e999f4b-2731-4f35-be87-4041f44d5c0c.gif)

Exercises before:
![Mittel (Exercises1)](https://user-images.githubusercontent.com/47261058/231386618-4402fcf0-4bec-4a40-aef0-9a587aa8b6fc.jpeg)
Exercises after:
![Mittel (Exercises2)](https://user-images.githubusercontent.com/47261058/231386624-22f4976c-0be8-45dc-bc7b-91bdfe607a46.jpeg)
